cmd/tag-set.go

```diff
@@ -0,0 +1,113 @@
+/*
+ * MinIO Client (C) 2020 MinIO, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package cmd
+
+import (
+	"errors"
+	"os"
+	"strconv"
+	"strings"
+
+	"github.com/fatih/color"
+	"github.com/minio/cli"
+	"github.com/minio/minio/pkg/console"
+)
+
+var tagSetCmd = cli.Command{
+	Name:   "set",
+	Usage:  "set/configure tags for objects",
+	Action: mainSetTag,
+	Before: setGlobalsFromContext,
+	Flags:  append(tagSetFlags, globalFlags...),
+	CustomHelpTemplate: `Name:
+	{{.HelpName}} - {{.Usage}}
+
+USAGE:
+  {{.HelpName}} [COMMAND FLAGS] TARGET
+
+FLAGS:
+  {{range .VisibleFlags}}{{.}}
+  {{end}}
+DESCRIPTION:
+   Show tags assigned to an object.
+
+EXAMPLES:
+  1. Assign the tag values to testobject in the bucket testbucket on alias s3.
+     {{.Prompt}} {{.HelpName}} s3/testbucket/testobject --tags "key1:value1" --tags "key2:value2" --tags "key3:value3"
+
+`,
+}
+
+var tagSetFlags = []cli.Flag{
+	cli.StringSliceFlag{
+		Name:  "tags",
+		Usage: "format '<key>:<value>'; multiple values allowed for multiple key/value pairs",
+	},
+}
+
+// Color scheme for set tag results
+func setTagSetColorScheme() {
+	console.SetColor(tagResultsSuccess, color.New(color.Bold, color.FgGreen))
+	console.SetColor(tagResultsFailure, color.New(color.Bold, color.FgRed))
+}
+
+func checkSetTagSyntax(ctx *cli.Context) {
+	if len(ctx.Args()) != 1 {
+		cli.ShowCommandHelp(ctx, "set")
+		os.Exit(globalErrorExitStatus)
+	}
+}
+
+func getTagMap(ctx *cli.Context) (map[string]string, error) {
+	ilmTagKVMap := make(map[string]string)
+	tagValues := ctx.StringSlice(strings.ToLower("tags"))
+	for tagIdx, tag := range tagValues {
+		key := splitStr(tag, ":", 2)[0]
```

Comment:
unable to just set `key` without `value` 
```
mc tag set s3/bucket/issue --tags "key:value" --tags "key1"
mc: <ERROR> error extracting tag argument(#2) key1. Unable to get tags to set.
```

Reply:
Change made, for the sake of uniformity key with empty value is to be entered as --tags "key:"